## Assignments

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

# Make a blank data frame.
df = pd.DataFrame()

# Populate it with data.
df['age'] = [28, 42, 27, 24, 35, 54, 35, 37]

## Summarizing data with Pandas

First let's set up some data and then we'll see how to summarize in Pandas.

In [2]:
import statistics
import numpy as np
import pandas as pd
%matplotlib inline

# Set up the data
data = [['Greg', 14], ['Marcia', 12], ['Peter', 11],
        ['Jan', 10], ['Bobby', 8], ['Oliver', 8], ['Cindy', 6]] 
  
df = pd.DataFrame(data, columns = ['Name', 'Age']) 

df.head()

Name  Age
0    Greg   14
1  Marcia   12
2   Peter   11
3     Jan   10
4   Bobby    8

### The `.describe()` method

In [3]:
df.describe()

Age
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

In [4]:
import statistics
import numpy as np
import pandas as pd
%matplotlib inline

# Set up the data
data = [['Greg', 14], ['Marcia', 12], ['Peter', 11],
        ['Jan', 10], ['Bobby', 8], ['Oliver', 8], ['Cindy', 7]] 
  
df2 = pd.DataFrame(data, columns = ['Name', 'Age']) 

In [5]:
df2.describe()

Age
count   7.000000
mean   10.000000
std     2.516611
min     7.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

In [6]:
import statistics
import numpy as np
import pandas as pd
%matplotlib inline

# Set up the data
data = [['Greg', 14], ['Marcia', 12], ['Peter', 11],
        ['Jan', 10], ['Bobby', 8], ['Jessica', 1], ['Cindy', 7]] 
  
df3 = pd.DataFrame(data, columns = ['Name', 'Age']) 

In [7]:
df3.describe()

Age
count   7.000000
mean    9.000000
std     4.242641
min     1.000000
25%     7.500000
50%    10.000000
75%    11.500000
max    14.000000

## Assignments

Now that we have introduced some tools for describing populations, let's try them out. First do these drills by hand, then use the Python code we've provided in the previous assignments to check your work. Keep track of your work in a Google document or markdown file that you can submit below and share with your mentor.

 1. Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch.  Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?
 2. Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?
 3. Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?
 4. Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again.  Does this change your choice of central tendency or variance estimation methods?
 5. On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show.  The answers were:
    TV Guide            20% fans
    Entertainment Weekly    23% fans
    Pop Culture Today       17% fans
    SciPhi Phanatic     5% fans

 Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

Discuss your answer to each of these questions, along with your code, with your mentor.

When you've given it a try, you can find a solution [here](https://github.com/Thinkful-Ed/data-201-resources/blob/master/solutions/Prep%20course/3.1.4.ipynb).

1.
Mean = 9.86
Median = 10
Mode = 8
Variance = 7.45
Standard Deviation = 2.73
Standard Error = 1.03

2.
I would choose the mean, since there doesn't seem to be any outliers in the dataset.

3.
After Cindy's birthday, the mean, standard deviation, standard deviation, standard error and minimum changed.

4.
I would use the median instead, because there is an outlier in the dataset.

5.
I would estimate that 16.25% of Americans were fans of the show.